### PR TITLE
version 1.0.0

### PR DESCRIPTION
**Pull request by armanso**

### CODE REVIEW

File: src/appstore-reviews.ts [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+        if (regions.length == 0) {
+            throw "At least one region should be define in configuration"
+        }
+
+        const requests = (regions as string[]).map(region => this.fetchAppStoreReviews(id, config.pageRange || 5, region))
+        const result = await Promise.all(requests)
+
+        let reviews: Review[] = []
+        var newReviews = reviews.concat(...result).filter((review) => {
+            if (publishedReviews[id] !== undefined && publishedReviews[id].indexOf(review.id) >= 0) {
```
Comment (mpourismaiel) [inline src/appstore-reviews.ts:28]: The if here can be replaced by return.
return publishedReviews[id]...

File: src/appstore-reviews.ts [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+        })
+
+        var newReviewsMap: PublishedReviews = {}
+        newReviewsMap[id] = newReviews.map(this.mapReviewId)
+
+        return {
+            messages: newReviews.map(review => {
+                if (config.generateMessageFromReview !== undefined) {
+                    return config.generateMessageFromReview.call(this, review, appInformation, config)
+                } else {
```
Comment (mpourismaiel) [inline src/appstore-reviews.ts:41]: This is more code style but removing else would make code more readable imho.

File: src/appstore-reviews.ts [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+                return []
+            }
+
+            return entries
+                .filter(this.isAppInformationEntry)
+                .reverse()
+                .map((review) => this.parseAppStoreReview(review, region))
+
+        } catch {
+            return []
```
Comment (mpourismaiel) [inline src/appstore-reviews.ts:99]: Logging the error that caused the exception would help debugging later on.

File: src/playstore-reviews.ts [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+        const appInformation = await app({ appId: id })
+        const result = await this.fetchPlayStoreReviews(id, config.publisherKey)
+
+        let reviews: Review[] = []
+        var newReviews = reviews.concat(...result).filter((review) => {
+            if (publishedReviews[id] !== undefined && publishedReviews[id].indexOf(review.id) >= 0) {
+                return false
+            }
+            return true
```
Comment (mpourismaiel) [inline src/playstore-reviews.ts:23]: Same as above. Returning boolean in if conditions is generally discouraged.

File: src/appstore-reviews.ts (hunk 1 above)
Comment (armanso) [reply src/appstore-reviews.ts]: Agree

File: src/appstore-reviews.ts (hunk 2 above)
Comment (armanso) [reply src/appstore-reviews.ts]: fixed

File: src/appstore-reviews.ts (hunk 3 above)
Comment (armanso) [reply src/appstore-reviews.ts:97]: Opps, I missed it

File: src/playstore-reviews.ts (hunk 4 above)
Comment (armanso) [reply src/playstore-reviews.ts]: Fixed.

**Comment by armanso** [comment]
> The code is fabulous. Just a few notes here and there. Also, use of let and const instead of var is encouraged.
> 
> Sorry for the delay.

Thanks bro.

**Pull request merged**